### DISCUSSION TITLE
Windows build issues

### DISCUSSION THREAD

**Discussion by akbyrd** [bulletphysics/bullet3#3761] [General]
Using the quickstart guide I've run into a handful of issues building the latest Bullet release, 2.87.

Windows 10 (1803)
Visual Studio 2015 Update 3
Used build_visual_studio_without_pybullet_vr.bat
Building App_BulletExampleBrowser

- The guide says to run Bullet/build/vs2010.bat, but that file doesn't exist.
- `Error C2664 'HMODULE LoadLibraryW(LPCWSTR)': cannot convert argument 1 from 'const char *' to 'LPCWSTR'`
Happens in a bunch of projects. B3_DYNLIB_OPEN is defined as LoadLibrary which evaluates to LoadLibraryW when Unicode is used, which it apparently is when generating projects withe the above batch file.
- `Error C1021 invalid preprocessor command 'warning'`
Comes from pathtool.h. There's a line beginning with #warning.
- `Error LNK1104 cannot open file 'openvr_api.lib'`
It looks like every premake batch file requires VR.

**Comment by akbyrd** [comment]
Other feedback

- btBulletCollisionCommon.h and btBulletDynamicsCommon.h aren't included in any projects. Following the Hello World section of the guide this causes a moment of 'uh, where does this file come from?' and a couple extra steps to track it down.
- autoexp_ext.txt is mentioned in the user manual, but doesn't seem to actually exist.
- It appears there's no way to change between /MD and /MT from the command line? Static CRT linking is enabled by default and the only way I see to change it is to remove the StaticRuntime flag from premake4.lua
- It appears there's no way to add debug symbols to release builds. Even adding "Symbols" to the release flags doesn't copy the pdb files to the output directory.
- targetsuffix options don't appear to work. The output files have no suffixes.

**Comment by elect86** [comment]
Another feedback, apparently flag `"Unicode"` seems to be unsupported

If I comment [this line](https://github.com/bulletphysics/bullet3/blob/master/test/GwenOpenGLTest/premake4.lua#L5) out (`--`), premake works, otherwise it fails

Also `os.is()` is deprecated and it should be substituted by `os.target()`, but this is simply a warning
